template.php

```diff
@@ -102,6 +102,12 @@ function campaignion_foundation_preprocess_page(&$vars) {
   $teaser_blocks = ['views_actions-block', 'views_actions_promoted-block'];
   $content_blocks = array_keys($vars['page']['content_top'] + $vars['page']['content'] + $vars['page']['content_bottom']);
   $has_teasers = current_path() == 'node' || array_intersect($teaser_blocks, $content_blocks);
+  if ($vars['layout'] === 'cover-banner' && $has_sidebar) {
+    // Is anything left in the sidebar besides form blocks?
+    $has_sidebar = !empty(array_filter($vars['page']['sidebar_first'] + $vars['page']['sidebar_second'], function ($key) use ($vars) {
+      return strpos($key, '#') !== 0 && !in_array($key, $vars['form_blocks']);
+    }, ARRAY_FILTER_USE_KEY));
```

Comment:
The anonymous function needs a `use ($vars)` to be able to access the `$vars` array. PHP doesn’t do closures automatically.

Also it might be made more readable by using `element_children()`, then we don’t need to care about the `#`-magic.

Something like this might work:

```php
$sidebars = $vars['page']['sidebar_first'] + $vars['page']['sidebar_second'];
$has_sidebar = (bool) array_filter(element_children($sidebars), function ($key) use ($vars) {
  return !in_array($key, $vars['form_blocks']);
});
```

Reply:
It does `use ($var)`…? 

`element_children` is a good idea, thanks for suggesting it!

Reply:
> It does `use ($var)`…?

Ah ok. Somehow I did overlook that yesterday, sorry.